transports/transports_test.go

```diff
@@ -0,0 +1,21 @@
+package transports
+
+import (
+	"testing"
+
+	"github.com/containers/image/directory"
+	"github.com/containers/image/storage"
+	"github.com/containers/image/transports"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTransportFromImageName(t *testing.T) {
+	dirTransport := transports.TransportFromImageName("dir:/tmp/test")
+	assert.Equal(t, dirTransport.Name(), directory.Transport.Name())
+	storageTransport := transports.TransportFromImageName("containers-storage:ref:test")
+	assert.Equal(t, storageTransport.Name(), storage.Transport.Name())
+	badTransport := transports.TransportFromImageName("unknown:ref:test")
+	assert.Equal(t, badTransport, nil)
+	badTransport2 := transports.TransportFromImageName("unknown")
```

Comment:
It’s a bit ugly that “valid format, unknown transport” and “invalid format” result in the same return value.

OTOH, this is likely to be very rarely used, and fits the skopeo use pretty well… at the very least, the function documentation should say that it returns `nil` on invalid input as well.

Reply:
Done.